lib/_http_client.js

```diff
@@ -354,7 +354,7 @@ function socketCloseListener() {
       res.emit('aborted');
     }
     req.emit('close');
-    if (res.readable) {
+    if (!res.aborted && res.readable) {
```

Comment:
This only prevents our listener from being added and the `'close'` event from being emitted but the `'end'` event is still emitted no? I mean, any listeners added by the user will be still called.

Am I wrong?

Reply:
I think you are wrong:

```js
    if (!res.aborted && res.readable) {
      res.on('end', function() {
        this.emit('close');
      });
      res.push(null);
    } else {
      res.emit('close');
    }
```

It will emit `end´ because of `push(null)` now it will just emit `close` in the `else`

Reply:
Oh didn't see `push(null)`, it was hidden in the diff, sorry.